File: src/Modules/Administration/AddRank.cs
using Discord;
using Discord.Commands;
using System.Linq;
using System.Threading.Tasks;

namespace DEA.Modules.Administration
{
    public partial class Administration
    {
        [Command("AddRank")]
        [Summary("Adds a rank role for the DEA cash system.")]
        public async Task AddRank(double cashRequired, [Remainder] IRole rankRole)
        {
            if (rankRole.Position > Context.Guild.CurrentUser.Roles.OrderByDescending(x => x.Position).First().Position)
            {
                ReplyError($"DEA must be higher in the heigharhy than {rankRole.Mention}.");
            }

            if (Context.DbGuild.RankRoles.ElementCount == 0)
            {
                await _guildRepo.ModifyAsync(Context.DbGuild, x => x.RankRoles.Add(rankRole.Id.ToString(), cashRequired));
            }
            else
            {
                if (Context.DbGuild.RankRoles.Any(x => x.Name == rankRole.Id.ToString()))
                {
                    ReplyError("This role is already a rank role.");
                }
                if (cashRequired == 500)
                {
                    cashRequired = Context.DbGuild.RankRoles.OrderByDescending(x => x.Value).First().Value.AsDouble * 2;
                }
                if (Context.DbGuild.RankRoles.Any(x => (int)x.Value.AsDouble == (int)cashRequired))
                {
                    ReplyError("There is already a role set to that amount of cash required.");
                }

                await _guildRepo.ModifyAsync(Context.DbGuild, x => x.RankRoles.Add(rankRole.Id.ToString(), cashRequired));
            }

            await ReplyAsync($"You have successfully added the {rankRole.Mention} rank.");
        }
    }
}

File: src/commands/administration/AddRank.js
const db = require('../../database');
const patron = require('patron.js');
const util = require('../../utility');

class AddRank extends patron.Command {
  constructor() {
    super({
      name: 'addrank',
      aliases: ['setrank'],
      group: 'administration',
      description: 'Add a rank.',
      args: [
        new patron.Argument({
          name: 'role',
          key: 'role',
          type: 'role',
          example: 'Sicario'
        }),
        new patron.Argument({
          name: 'cashRequired',
          key: 'cashRequired',
          type: 'float',
          example: '500'
        })
      ]
    });
  }

  async run(msg, args) {
    if (args.role.comparePositionTo(msg.guild.me.highestRole) > 0) {
      return util.Messenger.replyError(msg.channel, msg.author, 'DEA must be higher in hierarchy than ' + args.role + '.');
    }

    if (msg.dbGuild.roles.rank.some((role) => role.id === args.role.id)) {
      return util.Messenger.replyError(msg.channel, msg.author, 'This rank role has already been set.');
    }

    await db.guildRepo.upsertGuild(msg.guild.id, new db.updates.Push('roles.rank', { id: args.role.id, cashRequired: Math.round(args.cashRequired) }));

    return util.Messenger.reply(msg.channel, msg.author, 'You have successfully added the rank role ' + args.role + ' with a cash required amount of ' + util.NumberUtil.USD(args.cashRequired) + '.');
  }
}

module.exports = new AddRank();
